src/middlewared/middlewared/plugins/domain_controller.py

```diff
@@ -21,6 +24,64 @@ class Config:
         domaincontroller['role'] = domaincontroller['role'].lower()
         return domaincontroller
 
+    @private
+    async def is_provisioned(self):
+        systemdataset = await self.middleware.call('systemdataset.config')
+        sysvol_path = f"{systemdataset['path']}/samba4"
+        provisioned = "org.ix.activedirectory:provisioned"
+        ret = False
+        with libzfs.ZFS() as zfs:
```

Comment:
You should not use libzfs from asyncio event loop. Please make this method synchronous.

Reply:
Actually lets please not use libzfs at all outside zfs.py plugin. All zfs operations should be contained to zfs plugin, main reason being libzfs is proving lets effective each day due to is lack of thread safety and blocking operations because of such. Having every ZFS usage to a single place will simplify things for us in the long run.